src/reactiveData.ml

```diff
@@ -454,6 +473,22 @@ module RList = struct
     in
     from_event (List.rev (value t)) e
 
+  let for_all fn data =
+    let f = fun acc -> function
+      | Set x ->
+        List.for_all fn x
+      | Patch updates ->
+        let rec iter acc = function
+          | [] -> acc
+          | X _ :: tl -> iter acc tl
+          | R _ :: tl -> iter (if acc then acc else List.for_all fn (value data)) tl
```

Comment:
I did not find anything better than `List.for_all fn (value data))` when removing a element which was not satisfying `fn`.